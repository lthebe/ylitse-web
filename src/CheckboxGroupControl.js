import React from 'react';
import PropTypes from 'prop-types';
import {
    FormControl, FormLabel, FormControlLabel, FormGroup,
} from 'material-ui/Form';
import Checkbox from 'material-ui/Checkbox';

const CheckboxGroupControl = ({
    label, options, disabled, required, className, onChange,
}) => (
    <FormControl
        component="fieldset"
        disabled={disabled}
        required={required}
        className={className}
    >
        <FormLabel component="legend">{label}</FormLabel>
        <FormGroup row>
            {Object.keys(options).map(key => (
                <FormControlLabel
                    key={key}
                    name={key}
                    label={key}
                    checked={options[key]}
                    control={<Checkbox onChange={onChange} />}
                />
            ))}

        </FormGroup>
    </FormControl>
);

CheckboxGroupControl.propTypes = {
    label: PropTypes.string.isRequired,
    // eslint-disable-next-line react/forbid-prop-types
    options: PropTypes.object.isRequired,
    disabled: PropTypes.bool,
    required: PropTypes.bool,
    className: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
};

CheckboxGroupControl.defaultProps = {
    disabled: false,
    required: false,
};

export default CheckboxGroupControl;
