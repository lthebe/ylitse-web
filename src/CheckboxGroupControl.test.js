import React from 'react';
import { shallow, mount } from 'enzyme';
import CheckboxGroupControl from './CheckboxGroupControl';

describe('CheckboxGroupControl', () => {
    const mockFunc = jest.fn();
    const languages = { Finnish: false };
    const props = {
        options: languages,
        className: '',
        label: 'Language',
        onChange: mockFunc,
    };

    test('renders correctly', () => {
        const wrapper = shallow(<CheckboxGroupControl {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('checkbox can be checked and unchecked', () => {
        const wrapper = mount(<CheckboxGroupControl {...props} />);
        const input = wrapper.find('input[name="Finnish"]');
        input.simulate('change');
        expect(mockFunc).toHaveBeenCalled();
    });
});
