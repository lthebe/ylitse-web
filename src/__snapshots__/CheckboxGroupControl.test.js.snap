// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroupControl renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckboxGroupControl
    className=""
    disabled={false}
    label="Language"
    onChange={[MockFunction]}
    options={
      Object {
        "Finnish": false,
      }
    }
    required={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(FormLabel)
          component="legend"
        >
          Language
        </WithStyles(FormLabel)>,
        <WithStyles(FormGroup)
          row={true}
        >
          <WithStyles(FormControlLabel)
            checked={false}
            control={
              <WithStyles(Checkbox)
                onChange={[MockFunction]}
              />
            }
            label="Finnish"
            name="Finnish"
          />
        </WithStyles(FormGroup)>,
      ],
      "className": "",
      "component": "fieldset",
      "disabled": false,
      "required": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Language",
          "component": "legend",
        },
        "ref": null,
        "rendered": "Language",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(FormControlLabel)
              checked={false}
              control={
                <WithStyles(Checkbox)
                  onChange={[MockFunction]}
                />
              }
              label="Finnish"
              name="Finnish"
            />,
          ],
          "row": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Finnish",
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "control": <WithStyles(Checkbox)
                onChange={[MockFunction]}
              />,
              "label": "Finnish",
              "name": "Finnish",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(FormLabel)
            component="legend"
          >
            Language
          </WithStyles(FormLabel)>,
          <WithStyles(FormGroup)
            row={true}
          >
            <WithStyles(FormControlLabel)
              checked={false}
              control={
                <WithStyles(Checkbox)
                  onChange={[MockFunction]}
                />
              }
              label="Finnish"
              name="Finnish"
            />
          </WithStyles(FormGroup)>,
        ],
        "className": "",
        "component": "fieldset",
        "disabled": false,
        "required": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Language",
            "component": "legend",
          },
          "ref": null,
          "rendered": "Language",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(FormControlLabel)
                checked={false}
                control={
                  <WithStyles(Checkbox)
                    onChange={[MockFunction]}
                  />
                }
                label="Finnish"
                name="Finnish"
              />,
            ],
            "row": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Finnish",
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "control": <WithStyles(Checkbox)
                  onChange={[MockFunction]}
                />,
                "label": "Finnish",
                "name": "Finnish",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
